src/main/java/com/shanebeestudios/skbee/elements/recipe/effects/EffBrewingRecipe.java

```diff
@@ -0,0 +1,93 @@
+@Since("INSERT VERSION")
+@RequiredPlugins("PaperMC 1.18+")
+public class EffBrewingRecipe extends Effect {
+
+    private static final boolean SUPPORTS_POTION_MIX = Skript.classExists("io.papermc.paper.potion.PotionMix");
+    private Expression<ItemStack> result;
+
+    private Expression<Object> keyID;
+    private Expression<Object> input;
+    private Expression<Object> ingredient;
+
+    static {
+        if (SUPPORTS_POTION_MIX) {
+            Skript.registerEffect(EffBrewingRecipe.class, "register [a] [new] brewing[[ ]stand] recipe for %itemstack% with input %itemstack/materialchoice% using %itemstack/materialchoice% (using|with (id|key)) %string/namespacedkey%");
+        }
+    }
+
+    @Override
+    public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+        result = (Expression<ItemStack>) exprs[0];
+        input = (Expression<Object>) exprs[1];
+        ingredient = (Expression<Object>) exprs[2];
+        keyID = (Expression<Object>) exprs[3];
+        return true;
+    }
+
+    @Override
+    protected void execute(Event event) {
+        NamespacedKey key = RecipeUtil.getKey(this.keyID.getSingle(event));
+        ItemStack result = this.result.getSingle(event);
+        RecipeChoice input = RecipeUtil.getRecipeChoice(this.input.getSingle(event));
+        RecipeChoice ingredient = RecipeUtil.getRecipeChoice(this.ingredient.getSingle(event));
+
+        if (key == null) {
+            RecipeUtil.error("Error registering brewing recipe - key is null");
+            RecipeUtil.error("Current Item: &6" + this.toString(event, true));
+            return;
+        } else if (result == null) {
+            RecipeUtil.error("Error registering brewing recipe - result is null");
+            RecipeUtil.error("Current Item: &6" + this.toString(event, true));
+            return;
+        } else if (input == null) {
+            RecipeUtil.error("Error registering brewing recipe - input is null");
+            RecipeUtil.error("Current Item: &6" + this.toString(event, true));
+            return;
+        } else if (ingredient == null) {
+            RecipeUtil.error("Error registering brewing recipe - ingredient is null");
+            RecipeUtil.error("Current Item: &6" + this.toString(event, true));
+            return;
+        }
+
+        PotionMix potionMix = new PotionMix(key, result, input, ingredient);
+        Bukkit.getPotionBrewer().removePotionMix(key);
+        Bukkit.getPotionBrewer().addPotionMix(potionMix);
+    }
+
+    @Override
+    public String toString(@Nullable Event event, boolean debug) {
+        return "register new brewing stand recipe for " + result.toString(event, debug)
+                + " with input " + input.toString(event, debug)
+                + " using " + ingredient.toString(event, debug)
+                + " with id " + keyID.toString(event, debug);
+    }
+}
```

Comment:
```suggestion
}

```
I dunno, but apparently it's common practice to have a blank line at the end. 
Don't ask me why, I just follow the rules 😆